test/pkg/test/framework.go

```diff
@@ -45,21 +49,28 @@ var (
 
 type Framework struct {
 	client.Client
-	LatticeClient services.Lattice
+	LatticeClient                       services.Lattice
+	TestCasesCreatedServiceNetworkNames map[string]bool //key: ServiceNetworkName; value: not in use, meaningless
+	TestCasesCreatedServiceNames        map[string]bool //key: ServiceName; value not in use, meaningless
+	TestCasesCreatedTargetGroupNames    map[string]bool //key: TargetGroupName; value: not in use, meaningless
 }
 
 func NewFramework(ctx context.Context) *Framework {
 	var scheme = scheme.Scheme
 	lo.Must0(v1beta1.Install(scheme))
 	lo.Must0(v1alpha1.Install(scheme))
 	framework := &Framework{
-		Client:        lo.Must(client.New(controllerruntime.GetConfigOrDie(), client.Options{Scheme: scheme})),
-		LatticeClient: services.NewDefaultLattice(session.Must(session.NewSession()), ""), // region is currently hardcoded
+		Client:                              lo.Must(client.New(controllerruntime.GetConfigOrDie(), client.Options{Scheme: scheme})),
+		LatticeClient:                       services.NewDefaultLattice(session.Must(session.NewSession()), config.Region), // region is currently hardcoded
+		TestCasesCreatedServiceNetworkNames: make(map[string]bool),
+		TestCasesCreatedServiceNames:        make(map[string]bool),
+		TestCasesCreatedTargetGroupNames:    make(map[string]bool),
 	}
+
 	SetDefaultEventuallyTimeout(180 * time.Second)
-	SetDefaultEventuallyPollingInterval(1 * time.Second)
-	BeforeEach(func() { framework.ExpectToBeClean(ctx) })
-	AfterSuite(func() { framework.ExpectToClean(ctx) })
+	SetDefaultEventuallyPollingInterval(10 * time.Second)
+	//AfterEach(func() { framework.ExpectToBeClean(ctx) })  //After each test finished, we sometime don't expect to resource to be cleaned
```

Comment:
After( or before) each test(`It()`) finished, we sometime don't expect to resource to be cleaned, it `BeforeEach` or `AfterEach ` indeed Needed, we could write them in the test code instead of framework

Reply:
In my opinion, it is better to clean up after each test.  It makes debugging easier, since each test is self-contained and will `not` get impacted from previous test. (TLDR, the order of test cases will not make result different)